driver-core/src/main/java/com/datastax/driver/core/policies/TokenAwarePolicy.java

```diff
@@ -176,10 +256,24 @@ public HostDistance distance(Host host) {
     String keyspace = statement.getKeyspace();
     if (keyspace == null) keyspace = loggedKeyspace;
 
-    if (partitionKey == null || keyspace == null)
-      return childPolicy.newQueryPlan(keyspace, statement);
+    final Set<Host> replicas;
+    if (partitionKey == null) {
+      String csTokens = getTokenRange(statement);
```

Comment:
why not returning array or some wrapper object to avoid string parsing?
(you have the data parsed, you encode them to a string and then it needs to be parsed again ... )

Reply:
Tuple is probably the best way from where I stand (or number array?)

Reply:
Initially, I was returning the TokenRange itself,. but that posed other problems. Notice that the HostConnectionPool.borrowConnection method requires the start and end Token objects, but TokenRange doesn't have any methods that return them. So, I would have to convert them to String and instantiate them again anyways. Also, notice that Strings are generic enough to accomodate any Token size, even when using other type of partitioner. So, trying to store this, for example, into a Long object could cause problems.